### PR TITLE
chore: migrate DropdownMenu

### PR DESCRIPTION
**Pull request by RobelTekle**
## DESCRIPTION
Migration of DropdownMenu component

### BREAKING CHANGES:
- The DropdownMenu component no longer accepts the `gutter` prop. This value is set to 4px by default. If you want to remove the gutter, you can use `withGutter` prop.

## SCREENSHOTS / SCREEN RECORDINGS

https://github.com/user-attachments/assets/47eeb343-e813-497b-adc4-b30c0f00946c

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [ ] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [ ] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/DropdownMenu/Arrow.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import * as Ariakit from '@ariakit/react'
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Arrow.tsx]: You could just import what you need:
```
import type { MenuArrowProps } from '@ariakit/react'
import { MenuArrow, useStoreState } from '@ariakit/react'
```

File: lib/src/components/DropdownMenu/Arrow.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import dropdownMenuStyles from './dropdown-menu.module.scss'
+
+const cx = classNames(dropdownMenuStyles)
+
+const transformMap: Record<string, string> = {
+  bottom: 'rotateZ(360deg)',
+  left: 'rotateZ(90deg)',
+  right: 'rotateZ(-90deg)',
+  top: 'rotateZ(180deg)',
+}
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Arrow.tsx]: You could remove this and then cf my suggestion

File: lib/src/components/DropdownMenu/Arrow.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+export const Arrow = forwardRef<HTMLDivElement, ArrowProps>(({ className, store }, ref) => {
+  const currentPlacement = Ariakit.useStoreState(store, 'currentPlacement')
+  const [placement] = currentPlacement.split('-')
+
+  return (
+    <Ariakit.MenuArrow ref={ref} render={<div className={cx('arrow', className)} />}>
+      <svg
+        className={cx('arrow-item')}
+        style={{ '--dropdown-menu-transform': transformMap[placement] } as React.CSSProperties}
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Arrow.tsx]:
```suggestion
        className={cx('arrow-item', placement && `placement-${placement}`)}
```

Comment (cnairi) [reply lib/src/components/DropdownMenu/Arrow.tsx]: and then below arrow-item in `lib/src/components/DropdownMenu/dropdown-menu.module.scss`:
```
    .placement {
        &-bottom {
            --dropdown-menu-transform: rotateZ(360deg);
        }

        &-left {
            --dropdown-menu-transform: rotateZ(90deg);
        }

        &-right {
            --dropdown-menu-transform: rotateZ(-90deg);
        }

        &-top {
            --dropdown-menu-transform: rotateZ(180deg);
        }
    }
```

File: lib/src/components/DropdownMenu/Item.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import dropdownMenuStyles from './dropdown-menu.module.scss'
+
+const cx = classNames(dropdownMenuStyles)
+
+export type ItemProps = Ariakit.MenuItemProps
+
+export const Item = forwardRefWithAs<ItemProps, 'button'>(({ as: As, className, ...rest }, ref) => {
+  const Component = As || 'button'
+  return (
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Item.tsx]:
```suggestion
export const Item = forwardRefWithAs<ItemProps, 'button'>(
  ({ as: Component = 'button', className, ...rest }, ref) => {
    return (
```

File: lib/src/components/DropdownMenu/Item.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import * as Ariakit from '@ariakit/react'
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Item.tsx]:
```suggestion
import { MenuItem, type MenuItemProps } from '@ariakit/react'
```

File: lib/src/components/DropdownMenu/Separator.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import * as Ariakit from '@ariakit/react'
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Separator.tsx]:
```suggestion
import type { MenuSeparatorProps } from '@ariakit/react'
import { MenuSeparator } from '@ariakit/react'
```

File: lib/src/components/DropdownMenu/Trigger.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import * as Ariakit from '@ariakit/react'
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Trigger.tsx]:
```suggestion
import type { MenuButtonProps, MenuStore } from '@ariakit/react'
import { MenuButton } from '@ariakit/react'
```

File: lib/src/components/DropdownMenu/Trigger.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+import { forwardRefWithAs } from '@/utils'
+
+export type TriggerProps = TriggerOptions
+type TriggerOptions = { store: UseDropdownMenu }
+type UseDropdownMenu = Ariakit.MenuStore
+
+export const Trigger = forwardRefWithAs<TriggerProps, 'button'>(
+  ({ as: As, store, ...rest }, ref) => {
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/Trigger.tsx]: instead of `As` I would replace with Component, like:

`as: Component` and then `render={Component ? props => <Component {...props} /> : undefined}`

File: lib/src/components/DropdownMenu/dropdown-menu.module.scss [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+            color: var(--color-neutral-90);
+        }
+        &:hover {
+            background-color: var(--color-beige-30);
+            color: var(--color-neutral-90);
+        }
+        &[disabled] {
+            cursor: not-allowed;
+            color: var(--color-beige-40);
+            background-color: transparent;
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/dropdown-menu.module.scss]: background-color & color could be variables

```
background-color: var(--backgroundColor, transparent)
color: var(--color, --color-beige-70)

  &:focus {
            outline: none !important; /* important for firefox */
            --backgroundColor: var(--color-beige-30);
            --color: var(--color-neutral-90);
        }
```
and same for :hover & :disabled

File: lib/src/components/DropdownMenu/tests/index.test.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+const options = [
+  { label: 'one', value: 1 },
+  { label: 'two', value: 2 },
+  { label: 'three', value: 3 },
+  { label: 'four', value: 4 },
+  { label: 'five', value: 5 },
+]
+
+const DropdownRenderer: React.FC<DropdownMenuOptions> = () => {
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/tests/index.test.tsx]:
```suggestion
const DropdownRenderer = () => {
```

File: lib/src/components/DropdownMenu/index.tsx [hunk 11]
```diff
@@ -0,0 +1,64 @@
+import * as Ariakit from '@ariakit/react'
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/index.tsx]:
```suggestion
import type { MenuProps, MenuStore, MenuStoreProps, MenuStoreState } from '@ariakit/react'
import { Menu, useStoreState } from '@ariakit/react'
```

File: lib/src/components/DropdownMenu/index.tsx [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+const cx = classNames(dropdownMenuStyles)
+
+const DEFAULT_GUTTER = 4
+
+export interface DropdownMenuOptions extends Omit<Ariakit.MenuProps, 'gutter'> {
+  dataTestId?: string
+  innerProps?: React.ComponentProps<'div'>
+  withGutter?: boolean
+}
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/index.tsx]: Maybe you could gather all your Dropdown related types in a types.ts file

File: lib/src/components/DropdownMenu/index.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    const gutter = withGutter && !arrowElement ? DEFAULT_GUTTER : 0
+
+    const { className: innerClassName, ...otherInnerProps } = innerProps
+
+    return (
+      <Ariakit.Menu
+        aria-label="dropdown-menu"
+        className={className}
+        data-testid={dataTestId}
+        gutter={gutter}
```
Comment (cnairi) [inline lib/src/components/DropdownMenu/index.tsx:35]: Why did we decide not to expose the gutter anymore? It wasn't used?

Comment (RobelTekle) [reply lib/src/components/DropdownMenu/index.tsx:35]: There was a problem with keeping it as it was:
https://wttj.slack.com/archives/C09CT2BSF0B/p1758554825485209

Comment (cnairi) [reply lib/src/components/DropdownMenu/index.tsx:35]: ➡️  https://github.com/WTTJ/welcome-ui/pull/2883/#discussion_r2371754005

**Comment by cnairi** [comment]
🔥 🔥 🔥 🔥 🔥
